Public/Src/Sandbox/Windows/DetoursServices/DetoursHelpers.cpp

```diff
@@ -687,33 +711,15 @@ bool ParseFileAccessManifest(
 
     g_manifestInternalDetoursErrorNotificationFileString = reinterpret_cast<const PManifestInternalDetoursErrorNotificationFileString>(&payloadBytes[offset]);
     g_manifestInternalDetoursErrorNotificationFileString->AssertValid();
-
 #ifdef _DEBUG
     offset += sizeof(uint32_t);
 #endif
-    uint32_t manifestInternalDetoursErrorNotificationFileSize = *(uint32_t*)(&payloadBytes[offset]);
-    offset += sizeof(uint32_t);
-
-    if (manifestInternalDetoursErrorNotificationFileSize != 0)
-    {
-        g_internalDetoursErrorNotificationFile = new wchar_t[manifestInternalDetoursErrorNotificationFileSize + 1]; // Reserve some space for \0 terminator at end.
-        ZeroMemory((void*)g_internalDetoursErrorNotificationFile, sizeof(wchar_t) * (manifestInternalDetoursErrorNotificationFileSize + 1));
-        memcpy_s((void*)g_internalDetoursErrorNotificationFile, sizeof(wchar_t) * (manifestInternalDetoursErrorNotificationFileSize + 1), (wchar_t*)(&payloadBytes[offset]), sizeof(wchar_t) * manifestInternalDetoursErrorNotificationFileSize);
-    }
-    
-    if (g_internalDetoursErrorNotificationFile != nullptr && g_internalDetoursErrorNotificationFile[0] == L'\0')
-    {
-        manifestInternalDetoursErrorNotificationFileSize = 0;
-        delete[] g_internalDetoursErrorNotificationFile;
-        g_internalDetoursErrorNotificationFile = nullptr;
-    }

```

Comment:
Thanks for refactoring this.

Reply:
yw. I'll follow up after this PR with a refactor to convert this code to be more like the MacOS code, using a byte *cursor and simplify a lot of this code. Didn't want to mingle two major changes.

---
In reply to: [282689460](https://github.com/microsoft/BuildXL/pull/279#discussion_r282689460) [](ancestors = 282689460)